assets/src/js/admin/reports/components/mini-chart/index.js

```diff
@@ -1,81 +1,85 @@
-        <path fill-rule="evenodd" clip-rule="evenodd" d="M7 13C10.3137 13 13 10.3137 13 7C13 3.68629 10.3137 1 7 1C3.68629 1 1 3.68629 1 7C1 10.3137 3.68629 13 7 13Z" fill="#D75A4B" fill-opacity="0.3"/>
-        <mask id="mask0" mask-type="alpha" maskUnits="userSpaceOnUse" x="3" y="5" width="8" height="5">
-        <path fill-rule="evenodd" clip-rule="evenodd" d="M9.6775 5.01074L7 7.68241L4.3225 5.01074L3.5 5.83324L7 9.33324L10.5 5.83324L9.6775 5.01074Z" fill="white"/>
-        </mask>
-        <g mask="url(#mask0)">
-        <rect x="-7.5835" y="-7.5835" width="29.1667" height="29.1667" fill="#D75A4B"/>
-        </g>
-    </svg></div>
-    
-    const amount = getAmount(data)
-    const percentage = getPercentage(data)
-    const indicator = Math.sign(percentage) === -1 ? <Fragment>{down} <span style={{color: '#D75A4B'}}>{Math.abs(percentage) + '%'}</span></Fragment> : <Fragment>{up} <span style={{color: '#69B868'}}>{Math.abs(percentage) + '%'}</span></Fragment>
+	const [ highlightValue, setHighlightValue ] = useState( null );
+	const [ trend, setTrend ] = useState( null );
+	const [ indicator, setIndicator ] = useState( null );
 
-    const canvas = createRef()
-    const config = createConfig(data)
+	const canvas = createRef();
+	const config = createConfig( data );
 
-    useEffect(() => {
+	useEffect( () => {
+		const newHighlightValue = getHighlightValue( data );
+		const newTrend = getTrend( data );
+		const newIndicator = Math.sign( trend ) === -1 ? <Fragment>{ down } <span style={ { color: '#D75A4B' } }>{ Math.abs( trend ) + '%' }</span></Fragment> : <Fragment>{ up } <span style={ { color: '#69B868' } }>{ Math.abs( trend ) + '%' }</span></Fragment>;
 
-        const ctx = canvas.current.getContext('2d')
-        const chart = new ChartJS(ctx, config)
+		setHighlightValue( newHighlightValue );
+		setTrend( newTrend );
+		setIndicator( newIndicator );
 
-        return function cleanup() {
-            chart.destroy()
-        }
+		const ctx = canvas.current.getContext( '2d' );
+		const chart = new ChartJS( ctx, config );
 
-    }, [])
+		return function cleanup() {
+			chart.destroy();
+		};
+	}, [ data ] );
 
-
-    return (
-        <div className='givewp-mini-chart'>
-            <div className='header'>
-                <div className='title'>{title}</div>
-                <div className='indicator'>{indicator}</div>
-            </div>
-            <div className='content'>
-                <div className='amount'>{amount}</div>
-                <div className='chart'>
-                    <canvas width={100} height={40}  ref={canvas}></canvas>
-                </div>
-            </div>
-        </div>
-    )
-}
+	return (
+		<div className="givewp-mini-chart">
+			<div className="header">
+				<div className="title">{ title }</div>
+				<div className="indicator">{ indicator && ( indicator ) }</div>
```

Comment:
@henryholtgeerts what `{ indicator && ( indicator ) }` and `{ highlightValue && ( highlightValue ) }` this expression does

Reply:
Here I am checking that these variables have been defined before attempting to render them. https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator